react-js/react-components-and-hierarchies.md

```diff
@@ -0,0 +1,396 @@
+# Components & Hierarchies
+
+### Projected Time : 80-100 minutes
+- Lesson: 30-40 min
+- Guided Practice: 15-25 min
+- Independent Practice: 15-25 min
+
+
+### Prerequisites
+- [React JS Part 1]()
+- [Styling in JS](https://github.com/Techtonica/curriculum/blob/master/react-js/styling-react.md)
+- [React hooks](https://github.com/Techtonica/curriculum/blob/master/react-js/react-hooks.md)
+
+
+### Motivation
+ React is a User Interface (UI) library. Building a React application involves breaking up your app into smaller pieces called “components.” This make your code more readabel, manageable and scalable, as an app grows it can become difficult to get a sense of your overall component hierarchy and how components relate to each other. Sometimes, we want these components to communicate with each other and pass data among with each other. For this we use props and states. Sometimes, we would have to use 3rd party components to improve the the functionality or customization capability of out project
```

Comment:
typo: readabel

Please configure spellcheck in your editor as these are easy to catch.